src/item.cpp

```diff
@@ -5963,6 +5967,18 @@ bool item::is_money() const
     return ammo_types().count( ammotype( "money" ) );
 }
 
+bool item::is_software() const
+{
+    cata::optional<itype_id> optional_itype_usb_drive( "usb_drive" );
+    return type->default_container == optional_itype_usb_drive;
+}
+
+bool item::is_software_storage() const
+{
+    const itype_id itype_usb_drive( "usb_drive" );
+    return typeId() == itype_usb_drive;
+}
+
```

Comment:
I think the best we can do for now is to have `is_software_storage` be calculated by checking "does this have a software pocket" and `is_software` be calculated by "does the default container satisfy `is_software_storage`".

Alternatively, invent a new flag and add it to the flags for the `software` item from which all the others inherit and use that to determine software-ness.

Reply:
Okay.
```suggestion
bool item::is_software() const
{
    const itype_id &cont_value = type->default_container.value_or( "null" );
    if( !cont_value.is_null() ) {
        return item( cont_value ).is_software_storage();
    }
    return false;
}

bool item::is_software_storage() const
{
    return contents.has_pocket_type( item_pocket::pocket_type::SOFTWARE );
}
```
It is right?

Reply:
Looks reasonable.

Minor nit: I'd write the first function like:
```c++
    if( const cata_optional<itype_id> &cont = type->default_container ) {
        return item( cont ).is_software_storage();
    }
    return false;
```